brush-parser/src/parser.rs

```diff
@@ -254,8 +254,37 @@ peg::parser! {
             bang:bang()? seq:pipe_sequence() { ast::Pipeline { bang: bang.is_some(), seq } }
         rule bang() -> bool = specific_word("!") { true }
 
-        rule pipe_sequence() -> Vec<ast::Command> =
-            c:command() ++ (specific_operator("|") linebreak()) { c }
+        pub(crate) rule pipe_sequence() -> Vec<ast::Command> =
+            c:(c:command() r:input_output_pipe_extension()? {
+                let mut c = c;
+                if let Some(r) = r {
+                    match &mut c {
+                        ast::Command::Simple(c) => {
+                            let r = ast::CommandPrefixOrSuffixItem::IoRedirect(r);
+                            if let Some(suffix) = &mut c.suffix {
+                                suffix.0.push(r);
+                            } else {
+                                let v = vec![r];
+                                c.suffix = Some(ast::CommandSuffix(v));
+                            }
+                        }
+                        ast::Command::Compound(c, l) => {
+                            if let Some(r_list) = l {
+                                r_list.0.push(r);
+                            } else {
+                                let v = vec![r];
+                                *l = Some(ast::RedirectList(v));
+                            }
+
+                        }
+                        _ => { // TODO: syntax error!
```

Comment:
This begs the question of which kinds of commands can be followed by redirection operators. It looks like we already have a gap in `brush` with extended tests and others. For example, the following is accepted in `bash` but generates a syntax error in `brush`:

```
[[ -f path ]] >/dev/null
```

Function definitions can also have redirection lists, and it *seems* like `bash` will accept something like:

```bash
function foo() {
    echo hi
} |& wc -l
```

In other words, it seems like all the different command types should technically come with redirection list that we can add to?